library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -1,76 +1,69 @@
 package com.novoda.downloadmanager;
 
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
 import android.util.Log;
 
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.util.List;
 
-import static com.novoda.downloadmanager.DownloadBatchStatus.Status;
-
-class VersionOneToVersionTwoMigrator implements Migrator {
+class MigrationJob extends MigrationFutureWithCallbacks implements Runnable, DownloadMigrationService.MigrationFuture {
```

Comment:
I would avoid using inheritance wherever possible. You can use composition to achieve the same result here quite easily - I think if you have a `MigrationFutureWithCallbacks` as a field on this class you can call `onUpdate()` on it instead of using the inherited implementation.

Reply:
It's a mixin. I agree that composition is better. We could easily have it as a delegate instead.